testutil/network/basenet/network.go

```diff
@@ -0,0 +1,130 @@
+package basenet
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdknetwork "github.com/cosmos/cosmos-sdk/testutil/network"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/network"
+	"github.com/pokt-network/poktroll/testutil/testkeyring"
+)
+
+var _ network.InMemoryNetwork = (*BaseInMemoryNetwork)(nil)
+
+// BaseInMemoryNetwork is an "abstract" (i.e. partial) implementation, intended
+// to consolidate shared behavior between multiple ("concrete") InMemoryNetwork
+// implementations. These shared behaviors (methods) are accessible to any concrete
+// implementation which embeds BaseInMemoryNetwork.
+type BaseInMemoryNetwork struct {
+	Config                      network.InMemoryNetworkConfig
+	PreGeneratedAccountIterator *testkeyring.PreGeneratedAccountIterator
+	CosmosNetwork               *sdknetwork.Network
+
+	// lastValidatorSeqNumber stores the last (most recently generated) account sequence number.
+	// NB: explicitly NOT using atomic.Int32 as it's usage doesn't compose well with anonymous
```

Comment:
If this affects usage of the memory network struct in concurrent environments PUC - maybe link to the methods where it can be used concurrently?

Reply:
It should not as this is not an exported field. It should be accessed via `#GetLastValidatorTxSequenceNumber()` and `#NextValidatorTxSequenceNumber()`.